### ISSUE TITLE
Improve the color contrast for next/previous months days under date picker

### ISSUE THREAD

**Issue by RuiminLi98** [Esri/calcite-design-system#11644]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

**Context**
We have an accessibility ticket related to the date picker (https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/55500)

**Proposal**
We would like to improve the color contrast(legibility) for the next/previous months days instead of hiding them. 
(https://teams.microsoft.com/l/message/19:b405489d3ffb4001a914073af59b5e24@thread.skype/1740638686741?tenantId=aee6e3c9-711e-4c7c-bd27-04f2307db20d&groupId=56fae21a-9407-4943-859f-a9bfcf0bbad3&parentMessageId=1740520615282&teamName=Calcite%20Design%20System&channelName=Get%20Help%20-%20Design%20and%20Figma&createdTime=1740638686741)

### Actual Behavior

Currently, when we use date picker, the days from next/previous months having a color contrast is less than 3 which caused accessibility issue.

### Expected Behavior

We should have the days from next/previous months having a color contrast at least 3 to pass our accessibility check.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/date-picker/

### Reproduction Steps

2. Locate the date picker component.
2. Inspect it with Chrome DevTools.
3. In the Styles tab, copy the text color and background color.
4. Use a calculator such as https://www.levelaccess.com/color-contrast-checker-new/ to determine the contrast ratio.
5. Notice that the contrast ratio is below the required value for this size of text.

### Reproduction Version

11.4

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise

**Comment by ashetland** [comment]
Updated description with design updates from the Calcite design team.
